packages/npm-dependencies-validation/src/utils/npmUtil.ts

```diff
@@ -0,0 +1,26 @@
+import { spawn } from "child_process";
+
+export function getNPM(): string {
+  return /^win/.test(process.platform) ? "npm.cmd" : "npm";
+}
+
+export function spawnCommand(
+  commandArgs: string[],
+  workingDir: string
+): Promise<any> {
+  return new Promise((resolve, reject) => {
+    const command = spawn(getNPM(), [...commandArgs, "--json"], {
```

Comment:
note the `"--json"` already assumes this is an `invokeNPMListCommand` rather than a **general** invoke npm command. 

So maybe continue this trend make the three variants which abstract away the specifics of each command

Reply:
- is there a promise version of `spawn`?
- are there 3rd party libraries that perform this wrapping?

Reply:
there is no official async version of spawn.
One 3rd party lib , that I found, is not looking good